service/controller/resource/namespace/resource.go

```diff
@@ -49,7 +49,8 @@ func getObjectMeta(v interface{}) (metav1.ObjectMeta, error) {
 	}
 
 	return metav1.ObjectMeta{
-		Name: key.Namespace(cluster),
+		Name:   key.Namespace(cluster),
+		Labels: key.NamespaceLabels(),
```

Comment:
Nice one, I like it.
Did you test upgrade? I assume adding labels should be fine, but I remember some weirdness when changing namespace resources, so just double-checking.